### PR TITLE
Renames `TensExpr/Add/Mul` to `TensorExpr/Add/Mul`

### LINKED ISSUE #17508: Rename TensExpr, TensAdd and TensMul
A very simple issue that for naming consistency `TensExpr` should be `TensorExpr`, `TensAdd` should be `TensorAdd` and `TensMul` should be `TensorMul`.

### PR DESCRIPTION
**Pull request by drybalka**
#### References to other Issues or PRs
Fixes #17508 

#### Brief description of what is fixed or changed
Trivial change of names.

#### Release Notes

- tensor
  - `TensExpr/Add/Mul` classes renamed to `TensorExpr/Add/Mul`

#### Backwards Compatibility Breaks and Deprecations
- tensor
  -  `TensExpr/Add/Mul` classes renamed to `TensorExpr/Add/Mul`

### CODE REVIEW

**Comment by sylee957** [comment]
Is there a reason for renaming down those or encountered any problems?

File: sympy/tensor/tensor.py [hunk 1]
```diff
@@ -1815,7 +1815,7 @@ def tensor_heads(s, index_types, symmetry=None, comm=0):
     return thlist
 
 
-class TensExpr(Expr):
+class TensorExpr(Expr):
```
Comment (oscargus) [inline sympy/tensor/tensor.py]: I think that we should deprecate TensExpr etc.

**Comment by drybalka** [comment]
> Is there a reason for renaming down those or encountered any problems?

Not really, it is just a bit irritating to have `TensorIndex` and `TensorElement`, but `TensExpr` and `TensAdd` in the same module. Besides, similar methods for the matrix module are called `MatrixAdd/Mul` and not `MatrAdd`.

**Comment by sylee957** [comment]
Matrix module also have similar inconsistency
`MatrixExpr`, `MatrixSymbol`, `MatrixElement` vs `MatAdd`, `MatMul`, `MatPow`

**Comment by drybalka** [comment]
> Matrix module also have similar inconsistency
> `MatrixExpr`, `MatrixSymbol`, `MatrixElement` vs `MatAdd`, `MatMul`, `MatPow`

Well, maybe it needs fixing too. There is even a mention of `MatExpr` somewhere in matrices tests, so maybe I am not the only one confused.

File: sympy/tensor/tensor.py (hunk 1 above)
Comment (drybalka) [reply sympy/tensor/tensor.py]: You're right, I added deprecation messages for old names.

Comment (oscargus) [reply sympy/tensor/tensor.py]: Great, but I think you should return a `TensorExpr` object to not break user's code, but only print the deprecation message.

So

```
    def __new__(cls, *args, **kw_args):
        return TensorExpr(*args, **kw_args)
```
(or what the exact syntax is).

Comment (drybalka) [reply sympy/tensor/tensor.py]: No problem, I added the constructor you proposed and the staticmethods to `TensExpr` and others.

**Comment by Upabjojr** [comment]
Renaming stuff is can be helpful in some cases. I don't see the need here.

**Comment by drybalka** [comment]
> Renaming stuff is can be helpful in some cases. I don't see the need here.

You're probably right, there is no need. I am closing this PR.

**Pull request closed without merging**